MHVLib/Methods/MHVRestRequest.h

```diff
@@ -0,0 +1,53 @@
+//
+//  MHVRestRequest.h
+//  MHVLib
+//
+//  Created by Michael Burford on 5/22/17.
+//  Copyright © 2017 Microsoft Corporation. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "MHVHttpServiceOperationProtocol.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSInteger, MHVRestRequestType)
+{
+    MHVRestRequestJSON,
+    //MHVRestRequestDownloadBlobData,
+    //MHVRestRequestDownloadBlobFile,
+    //MHVRestRequestUploadBlob,
+};
+
+@interface MHVRestRequest : NSObject <MHVHttpServiceOperationProtocol>
+
+@property (nonatomic, assign, readonly)           MHVRestRequestType restRequestType;
+
+@property (nonatomic, strong, readonly)           NSString      *path;
+@property (nonatomic, strong, readonly)           NSString      *method;
```

Comment:
api seemed odd there too.  Changed to httpMethod which matches what it is called in NSURL